### PR TITLE
Add Log Filter Options for Savon

### PR DESCRIPTION
**Pull request by lbblock2**
Added option to include filters for logs to Savon, to take advantage of https://github.com/savonrb/savon/blob/e76ecc00b84b998b012ecc33b55ca1edd443ec55/lib/savon/log_message.rb#L7 and thereby allow us to filter out passwords (as in https://github.com/savonrb/savon/blob/e76ecc00b84b998b012ecc33b55ca1edd443ec55/spec/savon/log_message_spec.rb#L44)

### CODE REVIEW

**Comment by lbblock2** [comment]
@jheth Are you the main maintainer of the Soapforce gem? How do I get a review for my PR? Thanks!

File: soapforce.gemspec [hunk 1]
```diff
@@ -21,6 +21,6 @@ Gem::Specification.new do |spec|
   spec.add_runtime_dependency "savon", ">= 2.3.0", '< 3.0.0'
 
   spec.add_development_dependency 'rspec', '>= 2.14.0', '< 4.0.0'
-  spec.add_development_dependency 'webmock', '>= 1.17.0', '< 2.0.0'
+  spec.add_development_dependency 'webmock', '2.3.2'
```
Comment (jheth) [inline soapforce.gemspec]: Why pin to a specific version? Can we do `>= 2.3.2`?

File: spec/lib/client_spec.rb [hunk 2]
```diff
@@ -319,7 +319,7 @@
 
   describe "#update" do
     before(:each) do
-      @body = "<tns:update><tns:sObjects><ins0:type>Opportunity</ins0:type><ins0:Id>003ABCDE</ins0:Id><tns:Name>SOAPForce Opportunity</tns:Name><tns:CloseDate>2013-08-12</tns:CloseDate><tns:StageName>Closed Won</tns:StageName></tns:sObjects></tns:update>"
+      @body = "<tns:update><tns:sObjects><ins0:type>Opportunity</ins0:type><tns:Id>003ABCDE</tns:Id><tns:Name>SOAPForce Opportunity</tns:Name><tns:CloseDate>2013-08-12</tns:CloseDate><tns:StageName>Closed Won</tns:StageName></tns:sObjects></tns:update>"
```
Comment (jheth) [inline spec/lib/client_spec.rb]: I didn't expect the namespace to change. Any idea why it did?

File: spec/lib/client_spec.rb [hunk 3]
```diff
@@ -18,7 +18,7 @@
       stub = stub_login_request({with_body: body})
       stub.to_return(:status => 200, :body => fixture("login_response")) #, :headers => {})
 
-      subject.login(username: 'testing', password: 'password_and_token')
+      subject.login(username: 'testing', password: 'password_and_token', filters: [])
```
Comment (jheth) [inline spec/lib/client_spec.rb]: Adding filters here doesn't look right. It's not used by the `login` method, it's only used from `initialize`.

File: spec/lib/client_spec.rb (hunk 2 above)
Comment (lbblock2) [reply spec/lib/client_spec.rb]: No, not really, but it looks like a Savon issue https://github.com/savonrb/savon/issues/830

**Comment by lbblock2** [comment]
@jheth Hi, any update on this? Thanks!

**Comment by jheth** [comment]
Hey @lbblock2 , sorry for the delay. Forgot to click merge.

**Pull request merged**